community/detectors/spring_cloud_gateway_cve_2022_22947/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202222947/Cve202222947VulnDetector.java

```diff
@@ -0,0 +1,240 @@
+  private static final String USER_AGENT = "TSUNAMI_SCANNER";
+  private static final String PLACEHOLDER = "ROUTER_TSUNAMI";
+
+  @VisibleForTesting
+  static final String VULN_DESCRIPTION =
+      "In spring cloud gateway versions prior to 3.1.1+ and 3.0.7+ , applications are vulnerable "
+          + "to a code injection attack when the Gateway Actuator endpoint is enabled, exposed and "
+          + "unsecured. A remote attacker could make a maliciously crafted request that could allow"
+          + " arbitrary remote execution on the remote host.";
+
+  private final HttpClient httpClient;
+  private final Clock utcClock;
+
+  @Inject
+  Cve202222947VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.httpClient = checkNotNull(httpClient);
+    this.utcClock = checkNotNull(utcClock);
+  }
+
+  private static boolean isWebServiceOrUnknownService(NetworkService networkService) {
+    return networkService.getServiceName().isEmpty()
+        || NetworkServiceUtils.isWebService(networkService)
+        || NetworkServiceUtils.getServiceName(networkService).equals("unknown")
+        || NetworkServiceUtils.getServiceName(networkService).equals("rtsp");
+  }
+
+  private static StringBuilder buildTarget(NetworkService networkService) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(buildWebApplicationRootUrl(networkService));
+    } else {
+      targetUrlBuilder
+          .append("http://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    return targetUrlBuilder;
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("CVE-2921-22205 starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(Cve202222947VulnDetector::isWebServiceOrUnknownService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private String createRouter(NetworkService networkService) throws IOException {
+    String router = UUID.randomUUID().toString().replace("-", "").substring(0, 6);
+    String url = buildTarget(networkService).append(ROUTES).append(router).toString();
+    String payload = new String(Base64.getDecoder().decode(POST_DATA)).replace(PLACEHOLDER, router);
+    HttpResponse httpResponse =
+        httpClient.send(
+            post(url)
+                .setHeaders(
+                    HttpHeaders.builder()
+                        .addHeader(CONTENT_TYPE, MediaType.JSON_UTF_8.toString())
+                        .addHeader(USER_AGENT, USER_AGENT)
```

Comment:
Please use `import static com.google.common.net.HttpHeaders.USER_AGENT;` as the user agent key for the header.

Reply:
Repair ok ~